connector/dialect.tdd

```diff
@@ -380,60 +151,66 @@
+      <formula>CAST(%1 AS DECIMAL)</formula>
+      <argument type='real' />
     </function>
     <function group='cast' name='FLOAT' return-type='real'>
-      <formula>CAST(%1 AS DOUBLE PRECISION)</formula>
-      <argument type='real' />
+      <formula>CAST(%1 AS DECIMAL)</formula>
+      <argument type='int' />
     </function>
     <function group='cast' name='FLOAT' return-type='real'>
-      <formula>CAST(%1 AS DOUBLE PRECISION)</formula>
+      <formula>CAST(%1 AS DECIMAL)</formula>
       <argument type='str' />
     </function>
     <function group='cast' name='FLOAT' return-type='real'>
-      <formula>((EXTRACT(EPOCH FROM %1) / ( 60 * 60 * 24)) + (365 * 70 + 17))</formula>
-      <argument type='datetime' />
-    </function>
-    <function group='cast' name='INT' return-type='int'>
-      <formula>(CASE
-	WHEN %1 THEN 1
-	WHEN NOT %1 THEN 0
-	ELSE NULL END)</formula>
+      <formula>CAST(%1 AS DECIMAL)</formula>
       <argument type='bool' />
     </function>
     <function group='cast' name='INT' return-type='int'>
-      <formula>CAST(TRUNC(%1) AS BIGINT)</formula>
+      <formula>CAST(%1 AS INT)</formula>
       <argument type='real' />
     </function>
     <function group='cast' name='INT' return-type='int'>
-      <formula>CAST(TRUNC(CAST(%1 AS DOUBLE PRECISION)) AS BIGINT)</formula>
+      <formula>CAST(%1 AS INT)</formula>
+      <argument type='int' />
+    </function>
+    <function group='cast' name='INT' return-type='int'>
+      <formula>CAST(%1 AS INT)</formula>
       <argument type='str' />
     </function>
     <function group='cast' name='INT' return-type='int'>
-      <formula>FLOOR((EXTRACT(EPOCH FROM %1) / ( 60 * 60 * 24)) + (365 * 70 + 17))</formula>
-      <argument type='datetime' />
+      <formula>CAST(%1 AS INT)</formula>
+      <argument type='bool' />
     </function>
     <function group='cast' name='STR' return-type='str'>
-      <formula>(CASE
-	WHEN %1 THEN &apos;1&apos;
-	WHEN NOT %1 THEN &apos;0&apos;
-	ELSE NULL END)</formula>
+      <formula>CAST(%1 AS STRING)</formula>
       <argument type='bool' />
     </function>
     <function group='cast' name='STR' return-type='str'>
-      <formula>CAST(%1 AS TEXT)</formula>
+      <formula>CAST(%1 AS STRING)</formula>
       <argument type='real' />
     </function>
     <function group='cast' name='STR' return-type='str'>
-      <formula>CAST(%1 AS TEXT)</formula>
+      <formula>CAST(%1 AS STRING)</formula>
+      <argument type='int' />
+    </function>
+    <function group='cast' name='STR' return-type='str'>
+      <formula>CAST(%1 AS STRING)</formula>
       <argument type='str' />
     </function>
     <function group='cast' name='STR' return-type='str'>
-      <formula>CAST(%1 AS TEXT)</formula>
+      <formula>CAST(%1 AS STRING)</formula>
+      <argument type='datetime' />
+    </function>
+    <function group='date' name='MONTH' return-type='int'>
+      <formula>CAST(EXTRACT(MONTH FROM %1) AS INTEGER)</formula>
```

Comment:
```suggestion
      <formula>EXTRACT(MONTH FROM %1)</formula>
```
Why do we need to cast the result to int? Doesn't `EXTRACT` return an int?

Reply:
I removed the cast to int